app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt

```diff
@@ -282,6 +282,42 @@ class DownloadCache(
         notifyChanges()
     }
 
+    /**
+     * Renames a manga in this cache.
+     *
+     * @param source the source id.
+     * @param mangaUniFile the manga's new directory.
+     * @param oldTitle the manga's old title.
+     * @param newTitle the manga's new title.
+     */
+    suspend fun renameManga(source: Long, mangaUniFile: UniFile, oldTitle: String, newTitle: String) {
+        rootDownloadsDirMutex.withLock {
+            val sourceDir = rootDownloadsDir.sourceDirs[source] ?: return
+            val oldMangaDirName = provider.getMangaDirName(oldTitle)
+            var oldChapterDirs: MutableSet<String>? = null
+            // Save the old name's cached chapter dirs
+            if (sourceDir.mangaDirs.containsKey(oldMangaDirName)) {
+                oldChapterDirs = sourceDir.mangaDirs[oldMangaDirName]?.chapterDirs
+                sourceDir.mangaDirs -= oldMangaDirName
+            }
+
+            // Retrieve/create the cached manga directory for new name
+            val newMangaDirName = provider.getMangaDirName(newTitle)
+            var mangaDir = sourceDir.mangaDirs[newMangaDirName]
+            if (mangaDir == null) {
+                mangaDir = MangaDirectory(mangaUniFile)
+                sourceDir.mangaDirs += newMangaDirName to mangaDir
+            }
+
+            // Add the old chapters to new name's cache
+            if (!oldChapterDirs.isNullOrEmpty()) {
+                mangaDir.chapterDirs += oldChapterDirs
+            }
```

Comment:
I've kinda forgotten how the `UniFile` internal works but if you make a new folder with the manga's old name do the downloads still work?

Reply:
On my device it looks like if you try to rename a folder to an existing folder, the one getting renamed gets a "(2)" at the end or something similar. I could try to reuse the folder if it exists, maybe, and move the files into it, but I'm slightly conerned it'll result in orphaned cbz's with (2) at the end.

Reply:
Thanks to ```                sourceDir.mangaDirs -= oldMangaDirName
```
I don't believe those will stick around.
Are you concerned about hanging `MangaDirectory` obj references?